File: src/main/java/mediathek/gui/actions/import_actions/ImportOldAbosAction.java
package mediathek.gui.actions.import_actions;

import javafx.application.Platform;
import javafx.scene.control.Alert;
import javafx.stage.FileChooser;
import mediathek.config.Konstanten;
import mediathek.tool.javafx.FXErrorDialog;

import javax.swing.*;
import java.awt.event.ActionEvent;

public class ImportOldAbosAction extends AbstractAction {
    public ImportOldAbosAction() {
        putValue(Action.NAME, "Alte Abos...");
        putValue(Action.SHORT_DESCRIPTION, "Ermöglicht den Import der Abos aus einer alten Konfigurationsdatei.");
    }
    @Override
    public void actionPerformed(ActionEvent e) {
        Platform.runLater(() -> {
            var fileChooser = new FileChooser();
            fileChooser.setTitle("Konfigurationsdatei öffnen");
            var selectedFile = fileChooser.showOpenDialog(null);
            if (selectedFile != null) {
                try {
                    var configReader = new OldConfigFileImporter();
                    var result = configReader.importAboBlacklist(selectedFile.getAbsolutePath(), true, false, false);
                    var alert = new ImportSettingsAlert(Alert.AlertType.INFORMATION);
                    String text = "Es wurden " + result.left + " Einträge importiert.";
                    alert.setContentText(text);
                    alert.showAndWait();
                } catch (Exception ex) {
                    Platform.runLater(() -> FXErrorDialog.showErrorDialog(Konstanten.PROGRAMMNAME,
                            "Fehler beim Importieren der Abos",
                            "Es trat ein Fehler beim Import der Abos auf.\n" +
                                    "Sollte dies häufiger auftreten kontaktieren Sie bitte das Entwicklerteam.",
                            ex));
                }
            } else {
                var alert = new ImportSettingsAlert(Alert.AlertType.WARNING);
                alert.setContentText("Der Import der Abos wurde abgebrochen.");
                alert.showAndWait();
            }
        });
    }
}

File: src/main/kotlin/mediathek/gui/actions/import_actions/ImportOldAbosAction.kt
package mediathek.gui.actions.import_actions

import mediathek.config.Konstanten
import mediathek.tool.FileDialogs.chooseLoadFileLocation
import mediathek.tool.SwingErrorDialog
import java.awt.event.ActionEvent
import javax.swing.AbstractAction
import javax.swing.JFrame
import javax.swing.JOptionPane

class ImportOldAbosAction(
    private val parent: JFrame,
) : AbstractAction() {
    init {
        putValue(NAME, "Alte Abos...")
        putValue(SHORT_DESCRIPTION, "Ermöglicht den Import der Abos aus einer alten Konfigurationsdatei.")
    }

    override fun actionPerformed(e: ActionEvent) {
        val selectedFile = chooseLoadFileLocation(parent, " Konfigurationsdatei öffnen", "")
        if (selectedFile != null) {
            try {
                val configReader = OldConfigFileImporter()
                val (foundAbos) = configReader.importAboBlacklist(selectedFile.absolutePath,
                                                                  importAbo = true,
                                                                  importBlacklist = false,
                                                                  importReplaceList = false)
                val text = "Es wurden $foundAbos Einträge importiert."
                JOptionPane.showMessageDialog(parent, text, Konstanten.PROGRAMMNAME, JOptionPane.INFORMATION_MESSAGE)
            }
            catch (ex: Exception) {
                val text = """
                    Es trat ein Fehler beim Import der Abos auf.
                    Sollte dies häufiger auftreten kontaktieren Sie bitte das Entwicklerteam.
                    """.trimIndent()
                SwingErrorDialog.showExceptionMessage(parent, text, ex)
            }
        } else {
            JOptionPane.showMessageDialog(parent, "Der Import wurde abgebrochen.", Konstanten.PROGRAMMNAME,
                                          JOptionPane.WARNING_MESSAGE)
        }
    }
}
